components/script/dom/create.rs

```diff
@@ -106,11 +109,42 @@ fn create_svg_element(name: QualName,
     }
 }
 
+// https://dom.spec.whatwg.org/#concept-create-element
+#[allow(unsafe_code)]
 fn create_html_element(name: QualName,
                        prefix: Option<Prefix>,
                        document: &Document,
                        creator: ElementCreator)
                        -> Root<Element> {
+    assert!(name.ns == ns!(html));
+
+    // Step 4
+    let definition = document.lookup_custom_element_definition(name.local.clone(), None);
+
+    if let Some(definition) = definition {
+        // TODO: Handle customized built-in elements. Relies on CE upgrades.
+        if definition.is_autonomous() {
+            let local_name = name.local.clone();
+            return match definition.create_element(document) {
+                Ok(element) => element,
+                Err(error) => {
+                    // Step 6. Recovering from exception.
+                    let global = document.global();
```

Comment:
Since GlobalScope::current now returns an option, you should be able to fallback to the given document's global if it's not available.